pykechain/models/widgets/widget.py

```diff
@@ -195,11 +191,68 @@ def update_associations(self, readable_models=None, writable_models=None, **kwar
         :raises APIError: when the associations could not be changed
         :raise IllegalArgumentError: when the list is not of the right type
         """
-        self._client.update_widget_associations(widget=self, readable_models=readable_models,
-                                                writable_models=writable_models, **kwargs)
+        self._client.update_widget_associations(
+            widget=self,
+            readable_models=readable_models,
+            writable_models=writable_models,
+            **kwargs
```

Comment:
see other comments on `part_instance` and `parent_part_instance` to add to the call as well.